src/libraries/System.Private.CoreLib/src/Resources/Strings.resx

```diff
@@ -3425,7 +3425,13 @@
   <data name="SetterHasNoParams" xml:space="preserve">
     <value>Setter must have parameters.</value>
   </data>
-  <data name="SpinLock_Exit_SynchronizationLockException" xml:space="preserve">
+  <data name="Lock_Enter_LockRecursionException" xml:space="preserve">
+    <value>The calling thread has reached the limit of recursively entering the lock.</value>
```

Comment:
This reads like each thread has its own recursion counter.
```suggestion
    <value>Lock cannot be recursively re-entered any further.</value>
```

Reply:
I went back and forth on the wording of this. I wanted to convey that there's some limit that was reached. From your suggested wording the first impression I'm getting is that Lock doesn't allow recursive enters. How about "The lock has reached the limit of recursive enters."?

Reply:
sounds good